### PR TITLE
♻️ Maintenance: Removed deprecation warnings and marks flaky tests

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Misc maintenance: Various minor issues in the repo. 

Mostly **warnings** displayed during test or execution

- 🐛 adds flaky to servicelib and mark ``test_context_aware_measure_parallelis`` @GitHK please check
- 🔨 fixes *DeprecationWarning*``The 'asyncio_mode' is 'legacy', switching to 'auto' for the sake of pytest-aiohttp backward compatibility. Please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file`` by adding configuration in``setup.cfg``
- 🔨 fixes [*PytestWarning*](https://github.com/ITISFoundation/osparc-simcore/pull/3085/commits/b500bbe0e769b32be947e371450a573f98a70ae6)
- 🔨 fixes [*UserWarning*  _client.py:2012: nclosed <httpx.AsyncClient](https://github.com/ITISFoundation/osparc-simcore/pull/3085/commits/b66a803b4087023a8e7737202b12ec8dea1c52c7)
- 🗑️ removes ``models_library.settings`` (now should use ``settings_library`` instead)
-   ⬆️ upgrades ``paramiko`` to avoid Warnings due to Blowfish deprecation in cryptography

|#|name|before|after|upgrade|count|packages|
|-|----|------|-----|-------|-----|--------|
|   1 | paramiko                  | 2.10.4          | 2.11.0     | *minor*    | 4 | director-v2🧪</br>postgres-database🧪</br>service-library🧪</br>web🧪 |

*Legend*: 
- ⬆️ base dependency (only services because packages are floating)
- 🧪 test dependency
- 🔧 tool dependency

### CODE REVIEW

File: services/catalog/src/simcore_service_catalog/services/director.py [hunk 1]
```diff
@@ -27,29 +27,38 @@
...
+            raise
 
-        app.state.director_api = director_client
+        app.state.director_api = client
 
 
 async def close_director(app: FastAPI) -> None:
-    if director_client := app.state.director_api:
-        await director_client.delete()
+    client: Optional[DirectorApi]
```
Comment (sanderegg) [inline services/catalog/src/simcore_service_catalog/services/director.py:59]: what is the point of declaring this variable?

File: services/director-v2/setup.cfg [hunk 2]
```diff
@@ -6,3 +6,6 @@ tag = False
 commit_args = --no-verify
 
 [bumpversion:file:VERSION]
+
+[tool:pytest]
+asyncio_mode = auto
```
Comment (sanderegg) [inline services/director-v2/setup.cfg:11]: why was this necessary? it is already in the makefile right?

**Review by sanderegg — APPROVED** [review]
thanks for taking care of this!

File: packages/service-library/tests/test_async_utils.py [hunk 3]
```diff
@@ -151,6 +151,7 @@ async def target_function(the_param: Any) -> None:
     assert str(excinfo.value).startswith(message) is True
 
 
+@pytest.mark.flaky(max_runs=3)  # FIXME: ANE pls review and remove this flaky mark
```
Comment (GitHK) [inline packages/service-library/tests/test_async_utils.py:154]: Have you got more insight about this?
This could be really really bad if it's the case. Why was it working before? What happened now?

File: services/catalog/src/simcore_service_catalog/services/director.py (hunk 1 above)
Comment (pcrespov) [reply services/catalog/src/simcore_service_catalog/services/director.py:59]: this way we can get autocompletion, search,  etc in vscode

File: services/director-v2/setup.cfg (hunk 2 above)
Comment (pcrespov) [reply services/director-v2/setup.cfg:11]: is it? I need to double check then. There was a warning in the tests ...

File: packages/service-library/tests/test_async_utils.py (hunk 3 above)
Comment (pcrespov) [reply packages/service-library/tests/test_async_utils.py:154]: No. It simply fails randomly. You might want to review actions history for this test or run it many times until it fails?

File: services/director-v2/setup.cfg (hunk 2 above)
Comment (pcrespov) [reply services/director-v2/setup.cfg:11]: But yes, I see that the scripts in ``ci`` and the ``makefile`` recipes incorporate this option in CLI
That is strange then. Perhaps I was running ``pytest`` manually via CLI.

File: packages/service-library/tests/test_async_utils.py (hunk 3 above)
Comment (GitHK) [reply packages/service-library/tests/test_async_utils.py:154]: OK I've added the below PR to fix the issue. The timing being waited was too low.

- https://github.com/pcrespov/osparc-simcore/pull/68

Feel free to merge or change.

**Review by GitHK — CHANGES_REQUESTED** [review]
Please consider my proposal
